src/App.tsx

```diff
@@ -1,26 +1,238 @@
+        );
+      });
+  };
+
+  const handleDeletingAllCompleted = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleDeletingTodo(todo.id);
+      }
+    });
+  };
+
+  const handleErrorNotifications = () => setError('');
+
+  const handleTogglingTodo = (todo: Todo) => {
+    handleManagingTodos(todo.id);
+
+    todoService
+      .updateTodo({ ...todo, completed: !todo.completed })
+      .then(updatedTodo => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === updatedTodo.id ? updatedTodo : todo,
+          ),
+        );
+      })
+      .catch(() => {
+        setError(Error.UnableUpdate);
+      })
+      .finally(() => {
+        if (inputRef.current) {
+          inputRef.current.disabled = false;
+          inputRef.current.focus();
+        }
+
+        setManagingTodos(currentIds =>
+          currentIds.filter(currentId => currentId !== todo.id),
+        );
+      });
+  };
+
+  const handleTogglingAllTodos = () => {
+    if (activeTodos.length > 0) {
+      activeTodos.forEach(handleTogglingTodo);
+    } else {
+      completedTodos.forEach(handleTogglingTodo);
+    }
+  };
+
+  const handleUpdatingTodo = (todo: Todo) => {
+    handleManagingTodos(todo.id);
+
+    todoService
+      .updateTodo(todo)
+      .then(updatedTodo => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === updatedTodo.id ? updatedTodo : todo,
+          ),
+        );
+      })
+      .catch(() => setError(Error.UnableUpdate))
+      .finally(() => {
+        if (inputRef.current) {
+          inputRef.current.disabled = false;
+          inputRef.current.focus();
+        }
+
+        setManagingTodos(currentTodos =>
+          currentTodos.filter(currentId => currentId !== todo.id),
+        );
+      });
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          inputRef={inputRef}
+          handleAddingTodo={handleAddingTodo}
+          title={title}
+          handleTypingTodo={handleTypingTodo}
+          handleTogglingAllTodos={handleTogglingAllTodos}
+        />
+
+        <TodoList
+          todos={todos}
+          tempTodo={tempTodo}
+          status={status}
+          handleDeletingTodo={handleDeletingTodo}
+          managingTodos={managingTodos}
+          handleUpdatingTodo={handleUpdatingTodo}
+          handleTogglingTodo={handleTogglingTodo}
+        />
+
+        {todos.length > 0 && (
```

Comment:
```suggestion
        {!!todos.length && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**